docs: agent cache config requirements (#13006)
hashicorp/vault: website/content/docs/agent/caching/index.mdx

- `when_inconsistent` `(string: optional)` - Set to one of `"fail"`, `"retry"`,
   or `"forward"`.

> -> **Note:** When the `cache` block is defined, either a
> [template][agent-template] or a [listener][agent-listener] must also be defined

Reviewer: nit suggestion:

```suggestion
-> **Note:** When the `cache` block is defined, at least one
[template][agent-template] or [listener][agent-listener] must also be defined
```

Author: Yep, that's better, thanks.
